frontend/public/components/app.jsx

```diff
@@ -132,13 +139,64 @@ const App = (props) => {
     }
   }, []);
 
+  const onCSPViolation = React.useCallback((event) => {
+    // eslint-disable-next-line no-console
+    console.warn('Content Security Policy violation detected', event);
+
+    // https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent
+    const cspReportObject = _.pick(event, [
+      // The URI of the resource that was blocked because it violates a policy.
+      'blockedURI',
+      // The column number in the document or worker at which the violation occurred.
+      'columnNumber',
+      // Whether the user agent is configured to enforce or just report the policy violation.
+      'disposition',
+      // The URI of the document or worker in which the violation occurred.
+      'documentURI',
+      // The directive that was violated.
+      'effectiveDirective',
+      // The line number in the document or worker at which the violation occurred.
+      'lineNumber',
+      // The policy whose enforcement caused the violation.
+      'originalPolicy',
+      // The URL for the referrer of the resources whose policy was violated, or null.
+      'referrer',
+      // A sample of the resource that caused the violation, usually the first 40 characters.
+      // This will only be populated if the resource is an inline script, event handler or style.
+      'sample',
+      // If the violation occurred as a result of a script, this will be the URL of the script.
+      'sourceFile',
+      // HTTP status code of the document or worker in which the violation occurred.
+      'statusCode',
+    ]);
+
+    // Attempt to infer Console plugin name from CSP report object
+    const pluginName =
+      getPluginNameFromResourceURL(cspReportObject.blockedURI) ||
+      getPluginNameFromResourceURL(cspReportObject.sourceFile);
+
+    if (pluginName) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `Content Security Policy violation seems to originate from plugin ${pluginName}`,
+      );
```

Comment:
when dynamic plugin triggers CSP violation, we can only see warning message `Content Security Policy violation detected`

``Content Security Policy violation seems to originate from plugin ${pluginName}`` doesn't print

![Screenshot 2024-09-27 at 4 46 18 PM](https://github.com/user-attachments/assets/e59e624b-33c4-4abc-9432-402989033618)

Reply:
@yapei @jhadvig 

When detecting whether the CSP violation originates from a dynamic plugin or from Console application code itself, we look at `blockedURI` and `sourceFile` properties - if their values start with `{consoleBaseURL}/api/plugins/` prefix, it means that the associated resource is fetched via Console Bridge server's plugin asset endpoint which allows us to infer the name of the dynamic plugin.

This is why the message is phrased "seems to originate from plugin" instead of just "originates from plugin". Any CSP violations which are due to `{consoleBaseURL}/api/plugins/...` requests are associated with dynamic plugins, _but any dynamic plugin can also make requests to other (non-Console) endpoints_.

If a plugin makes requests to non-Console endpoints (regardless of whether the requested service is inside or outside the cluster) it will trigger CSP violations _which we have no way of associating with that dynamic plugin_. These kind of CSP violations will be logged, but the "seems to originate from plugin" will not apply to them.

**Each dynamic plugin team should test their own plugin on a cluster and ensure there are no CSP violations that may be traced back to their plugin.**

In case a plugin really needs to make requests to non-Console endpoints then it should explicitly list such endpoints via `ConsolePlugin` custom resource ([CONSOLE-4265](https://issues.redhat.com/browse/CONSOLE-4265)).